calico_containers/tests/st/docker_host.py

```diff
@@ -15,16 +15,21 @@ def __init__(self, name):
         pwd = sh.pwd().stdout.rstrip()
         docker.run("--privileged", "-v", pwd+":/code", "--name", self.name, "-tid", "jpetazzo/dind")
```

Comment:
Mounting the code directory in the build container is a common paradigm for
build containers; it means we don't have to rebuild the container every
time we make a code change.

Not sure if the rest of the test code allows this rebuild-free fast path,
but it would be a nice feature to have.

(I'm a fan of not rebuilding docker containers if they are there, and using
a `make clean` to blow everything away in the rare case that requires a
rebuild, rather than trying to guarantee that `make test` will work every
time. The build server can just do `make clean test`.)

On Fri, May 29, 2015 at 11:14 AM, Alex Altair <notifications@github.com>
wrote:

> In calico_containers/tests/st/docker_host.py
> <https://github.com/Metaswitch/calico-docker/pull/147#discussion_r31349955>
> :
>
> > @@ -15,16 +15,21 @@ def __init__(self, name):
> >          pwd = sh.pwd().stdout.rstrip()
> >          docker.run("--privileged", "-v", pwd+":/code", "--name", self.name, "-tid", "jpetazzo/dind")
>
> Yeah, it kinda feels that way to me too. It should be noted though that we
> aren't using the host containers primarily for isolation, but instead for
> rapid testing of calico on multiple hosts.
>
> Seems like your suggestion is the direction that we should move in, but
> also that at this point I should probably start doing something other than
> improving the test framework now that we can easily write multi-host tests
> in python.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Metaswitch/calico-docker/pull/147/files#r31349955>.
>